addons/portal/utils.py

```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tools import consteq
+
+
+def validate_thread_with_hash_pid(thread, _hash, pid):
+    if not (_hash and pid):
```

Comment:
It's hard to distribute and to read big negations, I prefer `not _hash or not pid` but up yo tou.